core/model/src/main/java/org/eclipse/rdf4j/model/datatypes/XMLDatatypeUtil.java

```diff
@@ -158,6 +158,38 @@ public static boolean isDecimalDatatype(IRI datatype) {
 	 * @return true if it is an integer type
 	 */
 	public static boolean isIntegerDatatype(IRI datatype) {
+		if (datatype == null) {
+			return false;
+		}
+
+		if (XSD.INTEGER == datatype) {
+			return true;
+		} else if (XSD.LONG == datatype) {
+			return true;
+		} else if (XSD.INT == datatype) {
+			return true;
+		} else if (XSD.SHORT == datatype) {
+			return true;
+		} else if (XSD.BYTE == datatype) {
+			return true;
+		} else if (XSD.NON_POSITIVE_INTEGER == datatype) {
+			return true;
+		} else if (XSD.NEGATIVE_INTEGER == datatype) {
+			return true;
+		} else if (XSD.NON_NEGATIVE_INTEGER == datatype) {
+			return true;
+		} else if (XSD.POSITIVE_INTEGER == datatype) {
+			return true;
+		} else if (XSD.UNSIGNED_LONG == datatype) {
+			return true;
+		} else if (XSD.UNSIGNED_INT == datatype) {
+			return true;
+		} else if (XSD.UNSIGNED_SHORT == datatype) {
+			return true;
+		} else if (XSD.UNSIGNED_BYTE == datatype) {
+			return true;
+		}
```

Comment:
This is faster than looking up in a hashmap because we don't need to compute the hash first. This is also much easier for the JVM to optimize.

Reply:
How do you ensure that the passed datatype is exactly identical (and not only equal) to any of the constants?

Reply:
There is a fallback after this sequence of if/elseif that uses the HashSet. So this is faster if you get a match without being significantly slower if you don't get a match.